File: 100/SameTree.scala
/**
 * Definition for a binary tree node.
 * class TreeNode(var _value: Int) {
 *   var value: Int = _value
 *   var left: TreeNode = null
 *   var right: TreeNode = null
 * }
 */
object SameTree {
    def isSameTree(p: TreeNode, q: TreeNode): Boolean = {
        if (p != null && q != null) {
            if (p.value == q.value) isSameTree(p.left, q.left) && isSameTree(p.right, q.right)
            else false
        } else if (p == null && q == null) true else false
    }
}

File: 100/same-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSameTree(p *TreeNode, q *TreeNode) bool {
	if p != nil && q != nil {
		if p.Val == q.Val {
			return isSameTree(p.Left, q.Left) && isSameTree(p.Right, q.Right)
		} else {
			return false
		}
	} else if p == nil && q == nil {
		return true
	} else {
		return false
	}
}